packages/core/src/domain/session/sessionStoreOperations.spec.ts

```diff
@@ -236,18 +252,36 @@ const cookieOptions: CookieOptions = {}
         stubStorage.getSpy.and.returnValue(buildSessionString({ ...initialSession, lock: 'locked' }))
         processSessionStoreOperations({ process: processSpy, after: afterSpy }, sessionStoreStrategy)
 
-        const lockMaxTries = isLockEnabled() ? LOCK_MAX_TRIES : 0
-        const lockRetryDelay = isLockEnabled() ? LOCK_RETRY_DELAY : 0
+        const lockMaxTries = sessionStoreStrategy.lockOptions.enabled ? LOCK_MAX_TRIES : 0
+        const lockRetryDelay = sessionStoreStrategy.lockOptions.enabled ? LOCK_RETRY_DELAY : 0
 
         clock.tick(lockMaxTries * lockRetryDelay)
         expect(processSpy).not.toHaveBeenCalled()
         expect(afterSpy).not.toHaveBeenCalled()
-        expect(stubStorage.setSpy).not.toHaveBeenCalled()
 
         clock.cleanup()
       })
 
-      it('should execute cookie accesses in order', (done) => {
+      it('should remove the lock when reaching max number of retries', () => {
+        const clock = mockClock()
+
+        sessionStoreStrategy.persistSession(initialSession)
+        stubStorage.setSpy.calls.reset()
+
+        stubStorage.getSpy.and.returnValue(buildSessionString({ ...initialSession, lock: 'locked' }))
+        processSessionStoreOperations({ process: processSpy, after: afterSpy }, sessionStoreStrategy)
+
+        const lockMaxTries = sessionStoreStrategy.lockOptions.enabled ? LOCK_MAX_TRIES : 0
+        const lockRetryDelay = sessionStoreStrategy.lockOptions.enabled ? LOCK_RETRY_DELAY : 0
+
+        clock.tick(lockMaxTries * lockRetryDelay)
+
+        expect(toSessionState(stubStorage.currentValue(storageKey)).lock).toBeUndefined()
+
+        clock.cleanup()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏cleanups need to be in an `afterEach` hook (or maybe a `finally` statement?) else if the test is failing midway, it'll interfer with other tests.